blocks/v2-images-grid/v2-images-grid.js

```diff
@@ -0,0 +1,65 @@
+import { createOptimizedPicture } from '../../scripts/lib-franklin.js';
+import { createElement, removeEmptyTags, getTextLabel } from '../../scripts/common.js';
+
+const blockClassName = 'v2-images-grid';
+export default function decorate(block) {
+  const headings = [...block.querySelectorAll('h1, h2, h3, h4, h5, h6')];
+  headings.forEach((heading) => {
+    heading.classList.add(`${blockClassName}-title`, 'with-marker');
+    heading.parentElement.parentElement.remove();
+  });
+  // all items are inside a ul list with classname called 'v2-images-grid-items'
+  const ul = createElement('ul', { classes: `${blockClassName}-items` });
+  [...block.querySelectorAll(':scope > div > div')].forEach((cell) => {
+    // If cell contain any element, we add them in the ul
+    if (cell.childElementCount) {
+      const li = createElement('li', { classes: [`${blockClassName}-item`] });
+      li.append(...cell.childNodes);
+      ul.append(li);
+    }
+    cell.remove();
+  });
+  block.parentElement.prepend(...headings);
+  block.append(ul);
+
+  // give format to the first 4 list items
+  [...ul.children].forEach((li, idx) => {
+    if (idx < 4) {
+      const section = createElement('div');
+      const captionEle = li.querySelector('p:not(:has(*))');
```

Comment:
We should use `:has` in selectors because Firefox is still not supporting it 😞

Reply:
Maybe you can use the `getAllElWithChildren` function from `script.js`

Reply:
Thanks for this, it helped